crates/ln-dlc-node/src/ln_dlc_wallet.rs

```diff
@@ -52,10 +55,33 @@ impl LnDlcWallet {
         &self.ln_wallet
     }
 
-    pub(crate) fn tip(&self) -> anyhow::Result<(u32, BlockHeader)> {
+    pub(crate) fn tip(&self) -> Result<(u32, BlockHeader)> {
         let (height, header) = self.ln_wallet.get_tip()?;
         Ok((height, header))
     }
+
+    /// A list of on-chain transactions. Transactions are sorted with the most recent transactions
+    /// appearing first.
+    ///
+    /// This list won't be up-to-date unless the wallet has previously been synchronised with the
+    /// blockchain.
+    pub(crate) fn on_chain_transactions(&self) -> Result<Vec<TransactionDetails>> {
+        let mut txs = self
+            .ln_wallet
+            .get_wallet()
+            .context("Can't acquire lock on bdk_ldk wallet")?
+            .list_transactions(false)
+            .context("bar")?;
+
+        txs.sort_by(|a, b| {
+            b.confirmation_time
+                .as_ref()
+                .map(|t| t.height)
+                .cmp(&a.confirmation_time.as_ref().map(|t| t.height))
+        });
```

Comment:
This sorting is a bit superfluous for the task at hand because we later combine all the transactions, payments and trades and sort them again. But I thought it might be good to sort this regardless in case we ever need to consume the list of on-chain transactions separately.